It doesn't like pattern and explicit rules to be on the same line,
and it seems to be more picky when matching file (or really directory)
names with different numbers of trailing slashes.

Signed-off-by: Jan Beulich <jbeulich [at] novell>
Acked-by: Sam Ravnborg <sam [at] ravnborg>

---
firmware/Makefile | 2 +-
scripts/mkmakefile | 4 +++-
2 files changed, 4 insertions(+), 2 deletions(-)

--- linux-2.6.36-rc1/firmware/Makefile
+++ 2.6.36-rc1-make-3.82-fixes/firmware/Makefile
@@ -142,7 +142,7 @@ fw-shipped-$(CONFIG_YAM) += yam/1200.bin
 fw-shipped-all := $(fw-shipped-y) $(fw-shipped-m) $(fw-shipped-)

 # Directories which we _might_ need to create, so we have a rule for them.
-firmware-dirs := $(sort $(patsubst %,$(objtree)/$(obj)/%/,$(dir $(fw-external-y) $(fw-shipped-all))))
+firmware-dirs := $(sort $(addprefix $(objtree)/$(obj)/,$(dir $(fw-external-y) $(fw-shipped-all))))

 quiet_cmd_mkdir = MKDIR   $(patsubst $(objtree)/%,%,$@)
       cmd_mkdir = mkdir -p $@
--- linux-2.6.36-rc1/scripts/mkmakefile
+++ 2.6.36-rc1-make-3.82-fixes/scripts/mkmakefile
@@ -44,7 +44,9 @@ all:

 Makefile:;

-\$(all) %/: all
+\$(all): all
 	@:

+%/: all
+ @:
 EOF
